### DISCUSSION TITLE
How to change spacing between facets (rows/columns)

### DISCUSSION THREAD

**Discussion by mcp292** [vega/altair#3018] [Q&A] [answered]
# [MWE](https://altair-viz.github.io/gallery/grouped_bar_chart.html)

```py
import altair as alt
from vega_datasets import data


source = data.barley()

chart = alt.Chart(source).mark_bar().encode(
    x=alt.X('year:O', title="More than year"),
    y='sum(yield):Q',
    color='year:N',
    column='site:N'
)

chart.show()
```

![Overlapping x-axis titles](https://user-images.githubusercontent.com/49123398/231032718-ca4f2be7-8a54-4faa-b579-49f1149f09d3.png)

# Question

How can I space out the colums such that the labels no longer overlap?

**Comment by joelostblom** [comment, marked as answer]
There is a `spacing` parameter (although it might be nicer to make these vertical bar chart, add newlines, or rotate the labels):

```py

alt.Chart(source).mark_bar().encode(
    x=alt.X('year:O', title="More than year"),
    y='sum(yield):Q',
    color='year:N',
    column=alt.Column('site:N', spacing=50)
)
```

**Reply by mcp292** [reply]
So cool! Altair continues to impress!

Not sure I understood your suggestion, but I'm interested to hear it. I thought these were vertical bar charts. Do you mean non-faceted? What do you mean by adding newlines?

I understand the suggestion for rotating the label, that's actually a very clean fix! I'll have to look into how to do that!

**Reply by mcp292** [reply]
Okay so for rotate, something like this?

```py
import altair as alt
from vega_datasets import data


source = data.barley()

chart = alt.Chart(source).mark_bar().encode(
    x=alt.X(
        'year:O',
        axis=alt.Axis(
            title="More than year", titleAngle=-90, titleAlign="right")),
    y='sum(yield):Q',
    color='year:N',
    column='site:N'
)

chart.show()
```

![Rotated x-label](https://user-images.githubusercontent.com/49123398/231195907-ad3b579c-d038-4b15-84ff-941fec654cbe.png)
